backend/apps/content/bigo_complexity_challenge_plugin.py

```
@@ -30,24 +30,45 @@
+            "len": len, "list": list, "map": map, "max": max, "min": min,
+            "pow": pow, "range": range, "round": round, "set": set, "str": str,
+            "sum": sum, "tuple": tuple, "zip": zip,
+            "AssertionError": AssertionError, "Exception": Exception,
+            "ValueError": ValueError, "TypeError": TypeError,
+        },
+        "__random": random,
+    }
+
+    try:
+        # Pre-compile and execute the setup + function definition
+        exec(full_code, safe_globals)
```

Ruff (0.16.0)

[error] Use of `exec` detected

(S102)